### PR TITLE
Chore/408 make recents a query 1

### PR DESCRIPTION
**Pull request by PaulKlauser**
Description: 
First steps in separating our Category domain model from the database model. Renamed the DB model and stopped duplicating Category state in PresetsViewModel.

Recommend stepping through by-commit. PR is a little larger than I wanted, but I wanted to get the tests in for PresetsViewModel to add confidence in the review and hopefully make it a little easier.

Part of #408

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
+    override suspend fun getPhrasesForCategory(categoryId: String): List<Phrase> {
         return database.categoryDao().getCategoryWithPhrases(categoryId)?.phrases ?: listOf()
     }
 
     suspend fun addPhrase(phrase: Phrase) {
         database.phraseDao().insertPhrase(phrase)
     }
 
-    suspend fun addPhraseToRecents(phrase: Phrase) {
+    override suspend fun addPhraseToRecents(phrase: Phrase) {
```
Comment (CarolineLaw) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt:37]: do you still need to add phrase to recents if you changed he way we get them?

File: app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.willowtree.vocable.presets.IPresetsRepository
+import com.willowtree.vocable.room.CategoryDto
+import kotlinx.coroutines.flow.Flow
+
+class CategoriesUseCase(
+    private val presetsRepository: IPresetsRepository
+) {
+
+    fun categories(): Flow<List<CategoryDto>> {
```
Comment (ryanbrookepayne) [inline app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt:11]: Any reason you aren't just invoking this use case given that there's only one function.
Ref: https://chrynan.codes/invoking-usecases-the-kotlin-way/

```suggestion
    operator fun invoke: Flow<List<CategoryDto>> {
```

File: app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt (hunk 1 above)
Comment (PaulKlauser) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt:37]: Good question. I haven't changed the way we get them yet, I've only started differentiating the Category model that we store in the database (`CategoryDto`), from the Category domain model (`Category`)

Comment (PaulKlauser) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt:37]: Getting myself back up to speed here 😅 I haven't actually even introduced `Category` yet in this code, only started adding the seams to do so, in this case `CategoriesUseCase`, which will eventually transform the database model into the domain model

File: app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt (hunk 2 above)
Comment (PaulKlauser) [reply app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt:11]: Yea I've seen that pattern, do you feel strongly that it's more readable that way? I've always felt it's pretty 6-in-one-half-a-dozen-in-the-other, and it gets awkward as soon as you want to add a second function

Comment (ryanbrookepayne) [reply app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt:11]: I don't feel strongly about it. Just thought I'd point it out.

**Pull request merged**